lib/mail/message.rb

```diff
@@ -1496,7 +1496,7 @@ def mime_type
     # Returns the character set defined in the content type field
     def charset
       if @header
-        has_content_type? ? content_type_parameters['charset'] : @charset
```

Comment:
Note: `multipart` `content-type` [does not have a `charset` parameter](https://github.com/mikel/mail/pull/1145), so effectively `content_type_parameters['charset']` was returning a `nil`.